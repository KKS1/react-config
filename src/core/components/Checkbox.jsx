import React, { Component } from "react";

const Checkbox = props => {
  const { value, label, onChange, name, id } = props;
  return (
    <label className="c-checkbox">
      <input
        id={id}
        className="c-checkbox__input"
        name={name}
        type="checkbox"
        checked={typeof value === "undefined" ? false : value}
        value={value}
        onChange={event => props.onChange(event.target.checked)}
      />
      <i className="c-checkbox__icon" />
      <span className="c-checkbox__label">{label}</span>
    </label>
  );
};

export default Checkbox;
