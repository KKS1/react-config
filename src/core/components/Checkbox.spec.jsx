import React, { Component } from "react";
import Checkbox from "./Checkbox";

describe("Checkbox", () => {
  test("render_Checkbox", () => {
    const props = {
      id: "formOne_CheckboxOne",
      label: "Checkbox Label",
      value: true,
      onChange: val => {}
    };

    const wrapper = shallow(<Checkbox {...props} />);
    const instance = wrapper.getElement();
    expect(instance).toMatchSnapshot();
  });

  test("change_Checkbox_should_propogate_boolean", () => {
    let isChecked = false;
    const props = {
      id: "formOne_CheckboxOne",
      label: "Checkbox Label",
      onChange: val => (isChecked = val)
    };

    const wrapper = shallow(<Checkbox {...props} />);
    const instance = wrapper.getElement();

    wrapper.find("input").simulate("change", { target: { checked: true } });

    expect(isChecked).toBe(true);
  });
});
