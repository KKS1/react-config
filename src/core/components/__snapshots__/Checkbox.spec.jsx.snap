// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render_Checkbox 1`] = `
<label
  className="c-checkbox"
>
  <input
    checked={true}
    className="c-checkbox__input"
    id="formOne_CheckboxOne"
    onChange={[Function]}
    type="checkbox"
    value={true}
  />
  <i
    className="c-checkbox__icon"
  />
  <span
    className="c-checkbox__label"
  >
    Checkbox Label
  </span>
</label>
`;
